### ISSUE TITLE
NumbaTypeSafetyWarning when running demo code snippets

### ISSUE THREAD

**Issue by Andreasbirch** [ermshaua/claspy#13]
Hello,
I am trying to run the code snippets for the usage examples provided in the readme.

This is my code for the univariate time series:
```
from claspy.segmentation import BinaryClaSPSegmentation
from claspy.data_loader import load_tssb_dataset

dataset, window_size, true_cps, time_series = load_tssb_dataset(names=("CricketX")).iloc[0,:]
clasp = BinaryClaSPSegmentation()
clasp.fit_predict(time_series)
```
And for the multivariate time series:
```
from claspy.segmentation import BinaryClaSPSegmentation
from claspy.data_loader import load_has_dataset
dataset, window_size, true_cps, labels, time_series = load_has_dataset().iloc[107, :]
clasp = BinaryClaSPSegmentation()
clasp.fit_predict(time_series)
```
In both cases I get the same error:
```
/Users/me/PycharmProjects/TimeSeriesSegmenter/.venv/lib/python3.11/site-packages/claspy/nearest_neighbour.py:252: NumbaTypeSafetyWarning: unsafe cast from uint64 to int64. Precision may be lost.
  start, end = pranges[idx]
```

**Comment by ermshaua** [comment]
Do you get an error that stops your program or just a warning? Until now, this was just a warning. Haven't fixed it yet.

**Comment by Andreasbirch** [comment]
The warning stops the program. I was able to circumvent it by adding
```
from numba.core.errors import NumbaTypeSafetyWarning
import warnings
warnings.simplefilter(action='ignore', category=NumbaTypeSafetyWarning)
```
To the `nearest_neighbour.py` file.
I found the documentation for the solution [here](https://numba.pydata.org/numba-doc/dev/reference/deprecation.html)

**Comment by ermshaua** [comment]
Thanks! I'll have a look.

**Comment by ermshaua** [comment]
Should now be fixed in 0.2.4. Can you upgrade and verify?

**Comment by Andreasbirch** [comment]
Looks good - no warnings anymore! Thank you for your help.
